File: 2018 - 1/Estruturas de Dados/Alguns códigos/Estudos Ferrari Code/Fila/fila.c
#include "f

void create(struct Fila *f){
    f->inicio = 0;
    f->fim = 0;
    f->total = 0;


}
 void inserir(struct Fila *f, int ele){{
    f->elementos[f->fim] = elementos;
    f->fim = (f->fim + 1) % 10;
    f->total++;

}
int retirar(struct Fila *f){
    int el = f->inicio;
    f->inicio = (f->inicio + 1)%10;
    f->total--;
    return el;

}
int isEmpty(struct Fila f){
    return (f.total == 0);

}
int isFull(struct Fila f){
    return (f.total == 10);


}

File: 2018 - 1/Estruturas de Dados/Alguns códigos/_/Fila.cs
using System;


public class Fila{

private int[] valores;
private int primeiro; // inicio
private int ultimo; // fim
private int total;



public Fila()
{
    valores = new int[10];
    primeiro = 0;
    ultimo = 0;
    total = 0;

            




}


public void Inserir(int elemento)
{
    if (!Cheia())
    {
        valores[ultimo] = elemento;
        ultimo = (ultimo + 1) % valores.length;
        total++;
    }
    else { throw new RuntimeException("Fila cheia, nao da pra inserir"); };

}



public void Retirar(int* e)
{
    if (!Vazia())
    {
        e = valores[primeiro];
        primeiro = (primeiro + 1) % valores.length;
        total--;
    }

    else { throw new RuntimeException("Fila vazia, nao da pra remover "); };


}


public boolean Vazia()
{
    return (total == 0);


}

public boolean Cheia()
{
    return (total == (valores.length));


}

}
